_docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Service/WordpressApi.php

```diff
@@ -147,40 +209,67 @@ public function getAutocompleteContentOptions($search, $max_results) {
-      $response = $request->getBody()->getContents();
-      $api_results = json_decode($response);
-
-      foreach ($api_results as $api_result) {
-        $label = Html::escape($api_result->post_title) . ' (' . $api_result->ID . ')';
-        $results[] = [
-          'value' => $label,
-          'label' => $label,
-        ];
-
-        if (count($results) >= $max_results) {
-          // If we already have $max_results-many WP posts in $results, then we
-          // return $results.
-          return $results;
-        }
+    // Serve the cached Wordpress API data if available.
+    if ($cache = $this->cacheBin->get($cid)) {
+      $data = $cache->data;
+    }
+    // If this is uncached or the cache is invalid, fetch fresh data.
+    else {
+      try {
+        // Tries to retrieve WP posts from $feed_url using the $query params.
+        $request = $this->client->request('GET', $feed_url, ['query' => $query]);
+        $response = $request->getBody()->getContents();
+        $api_results = json_decode($response);
+        $data = $api_results;
+        // Persist this data to the wordpress_api cache bin for an hour.
+        $ttl = $this->configFactory->get('redhat_developers')->get('wordpressApi.ttl');
+        $this->cacheBin->set($cid, $data, time() + $ttl);
+      }
+      catch (\Exception $e) {
+        $this->loggerFactory->get('rhd_assemblies')->error(
+          "Exception while calling Wordpress API: @message", [
+            '@message' => $e->getMessage(),
+          ]
+        );
+
+        return [];
       }
-
-      // If no WP posts were returned in the response, return the empty $results
-      // array here.
-      return $results;
     }
-    catch (\Exception $e) {
-      \Drupal::logger('rhd_assemblies')->error(
-        "Exception while calling Wordpress API: @message", [
-          '@message' => $e->getMessage(),
-        ]
-      );
-
-      return [];
+
+    return $this->formatAutocompleteContentOptions($data, $max_results);
+  }
+
+  /**
+   * Returns formatted HTML output of the autocomplete options.
+   *
+   * @param string $api_results
+   *   The titles/results returned from the Wordpress API.
+   * @param int $max_results
+   *   The maximum number of results to provide in the autocomplete options.
+   *
+   * @return string
+   *   Returns the formatted HTML output.
+   */
+  public function formatAutocompleteContentOptions($api_results, $max_results) {
+    foreach ($api_results as $api_result) {
+      $label = Html::escape($api_result->post_title) . ' (' . $api_result->ID . ')';
+      $results[] = [
+        'value' => $label,
+        'label' => $label,
+      ];
+
+      if (count($results) >= $max_results) {
+        // If we already have $max_results-many WP posts in $results, then we
+        // return $results.
+        return $results;
+      }
     }
+
+    // If no WP posts were returned in the response, return the empty $results
+    // array here.
+    return $results;
```

Comment:
`$results` is set within a foreach;  will it have a value in the case that $api_results is empty?
How do we test this?